test/compatibility/common/compatibility_config.py

```
@@ -28,10 +28,36 @@
 
 # Common configuration for compatibility tests.
 
+import os
+import re
+from typing import Dict, List
+
+from compatibility_version import WTVersion
+
 # The branches we use for the testing. We support special branch name 'this' that refers to the
 # current branch. This is useful when debugging a compatibility issue on the current branch, but it
 # should not be enabled when testing on Evergreen.
-BRANCHES = ['develop', 'mongodb-7.1', 'mongodb-7.0', 'mongodb-6.3']
+
+# To make this branches compatitable with existing "compatibility_test_for_releases.sh", the
+# version is imported from the bash file of "meta/versions.sh"
+
+def version_extract(bash_script):
+    version_ret = {}
+    with open(bash_script, "r") as f:
+        lines = f.read().split('\n')
+    for line in lines:
+        match = re.match(r'export\s+([A-Z_]+)\s*=\s*"([^"]*)"', line)
+        if match:
+            name = match.group(1)
+            versions = match.group(2).split()
+            versions = [WTVersion(version) for version in versions]
+            version_ret[name] = [version for version in versions if version]
+    return version_ret
```

Raise an exception if you cannot parse the file, rather than silently ignoring lines that did not match. This could save us a lot of debugging time in the future.

---

I'll add a null dict judge. But in the future when rules are made, key existence checking can be made here.